sky/spot/recovery_strategy.py

```diff
@@ -159,6 +159,9 @@ def _launch(self, max_retry=3, raise_on_failure=True) -> Optional[float]:
             except exceptions.InvalidClusterNameError as e:
                 # The cluster name is too long.
                 raise exceptions.ResourcesUnavailableError(str(e)) from e
+            except exceptions.ClusterSetUpError as e:
+                # setup error
```

Comment:
nit: can we elaborate the comment a bit?

```suggestion
                # When the setup fails, do not retry, as it will probably fail again.
                # The exception will be handled in the caller and set the job to
                # FAILED.
```

Reply:
Added!